File: WeatherApp/app/src/main/java/com/example/android/Task03/ThreeDaysRecyclerViewAdapter.java
package com.example.android.Task03;

import android.content.Context;
import android.os.Build;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.RequiresApi;
import androidx.recyclerview.widget.RecyclerView;

import com.example.android.Task03.Retrofit.MainData;

import java.time.DayOfWeek;
import java.time.LocalDate;

public class ThreeDaysRecyclerViewAdapter extends RecyclerView.Adapter<ThreeDaysRecyclerViewAdapter.ViewHolder> {

    Context context;
    MainData[] mainData;
    int count;


    public ThreeDaysRecyclerViewAdapter(Context ct) {
        context = ct;
        count = 0;
    }

    public void setDataThreeDaysFragment(MainData[] values) {
        mainData = values;
        count = 0;
        notifyDataSetChanged();
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater layoutInflater = LayoutInflater.from(context);
        View view = layoutInflater.inflate(R.layout.three_days_list_item, parent, false);
        return new ViewHolder(view);
    }

    @RequiresApi(api = Build.VERSION_CODES.O)
    @Override
    public void onBindViewHolder(@NonNull ThreeDaysRecyclerViewAdapter.ViewHolder holder, int position) {


        if (mainData != null) {
            holder.imageView.setImageDrawable(MainActivity.weatherIcons.get(mainData[count].getWeathers()[0].getWeatherIcon()));
        }

        if (mainData != null) {
            LocalDate date = LocalDate.parse(mainData[count].getDate());
            String dayOfMonth = date.getDayOfMonth() + " " + date.getMonth();
            holder.textView2.setText(dayOfMonth);
        }
        if (mainData != null) {
            LocalDate date = LocalDate.parse(mainData[count].getDate());
            DayOfWeek dayOfWeek = date.getDayOfWeek();
            holder.textView1.setText(dayOfWeek.toString());
        }
        if (mainData != null) {
            holder.textView3.setText(mainData[count].getMainDataValues().getTemp());
            count += 8;
        }
    }

    @Override
    public int getItemCount() {
        return 3;
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        TextView textView1, textView2, textView3;
        ImageView imageView;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            textView1 = itemView.findViewById(R.id.day_of_week);
            textView2 = itemView.findViewById(R.id.day_of_month);
            textView3 = itemView.findViewById(R.id.temperature);
            imageView = itemView.findViewById(R.id.weather_condition);
        }
    }
}

File: WeatherApp Kotlin/app/src/main/java/com/example/android/weatherApp/ThreeDaysRecyclerViewAdapter.kt
package com.example.android.weatherApp

import android.content.Context
import android.os.Build
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.annotation.RequiresApi
import androidx.recyclerview.widget.RecyclerView
import com.example.android.weatherApp.retrofit.MainData
import java.time.LocalDate

class ThreeDaysRecyclerViewAdapter(private val context: Context?) : RecyclerView.Adapter<ThreeDaysRecyclerViewAdapter.ViewHolder>() {
    private var mainData: Array<MainData>? = null
    private var count = 0
    fun setDataThreeDaysFragment(values: Array<MainData>?) {
        mainData = values
        count = 0
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val layoutInflater = LayoutInflater.from(context)
        val view = layoutInflater.inflate(R.layout.three_days_list_item, parent, false)
        return ViewHolder(view)
    }

    @RequiresApi(api = Build.VERSION_CODES.O)
    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        if (mainData != null) {
            holder.imageView.setImageDrawable(MainActivity.weatherIcons[mainData!![count].weathers?.get(0)!!.weatherIcon])
        }
        if (mainData != null) {
            val date = LocalDate.parse(mainData!![count].getDate())
            val dayOfMonth = date.dayOfMonth.toString() + " " + date.month
            holder.textView2.text = dayOfMonth
        }
        if (mainData != null) {
            val date = LocalDate.parse(mainData!![count].getDate())
            val dayOfWeek = date.dayOfWeek
            holder.textView1.text = dayOfWeek.toString()
        }
        if (mainData != null) {
            holder.textView3.text = mainData!![count].mainDataValues.getTemp()
            count += 8
        }
    }

    override fun getItemCount(): Int {
        return 3
    }

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var textView1: TextView = itemView.findViewById(R.id.day_of_week)
        var textView2: TextView = itemView.findViewById(R.id.day_of_month)
        var textView3: TextView = itemView.findViewById(R.id.temperature)
        var imageView: ImageView = itemView.findViewById(R.id.weather_condition)

    }
}